### DISCUSSION TITLE
spdlog::get

### DISCUSSION THREAD

**Discussion by MimiXav** [gabime/spdlog#1929] [General]
Hi all,

I am using spdlog for an internal projecct, and it's cool, it's works very well ;)
But, I have an issue (incomprehension) with spdlog::get("loggerName");

When I create a logger by spdlog::stdout_color_mt("loggerName", spdlog::color_mode::automatic)
the function spdlog::get return Yes so perfect.

But unfortunatlly when I create the logger by spdlog::rotating_logger_mt("loggerName", .........;)
the function spdlog::get return no, so what is wrong?

Thanks for any help

Regards

**Comment by tt4g** [comment]
Loggers with the same name cannot be registered. The name of the logger must be changed.

**Reply by MimiXav** [reply]
Thanks for your answer,
But yes I agree with you, the main issue is 
Yes of course
when I create spdlog::rotating_logger_mt("MyloggerName", .........;) and

**Reply by MimiXav** [reply]
auto logger = spdlog::get(loggerName);
Say no

**Reply by tt4g** [reply]
Does the problem mean that `spdlog::get(loggerName)` returns an invalid `std::shared_ptr`?

**Reply by tt4g** [reply]
Please ignore this comment unless you are using a shared library.

If you are using spdlog with a shared library, `spdlog::get(loggerName)` will not work because the memory space is different between the application and the shared library.
Wiki: https://github.com/gabime/spdlog/wiki/How-to-use-spdlog-in-DLLs

**Reply by MimiXav** [reply]
Ok, I will see but Here my code
done spdlog::rotating_logger_mt("myLoggerName", rolling.filePattern, rolling.maxSize, rolling.maxFile);
and after   
            auto logger = spdlog::get("myLoggerName");
            if (!logger)
            {
              logger = createLogger("myLoggerName", logLevel);
            }
            return logger;

**Reply by tt4g** [reply]
Yes, this snippet will work.

**Reply by MimiXav** [reply]
Ok, I am working on this issue (in my side), I test something and I will send my feedback
Regards

**Reply by MimiXav** [reply]
So after working on my issue, it was the shared library issue !! ;) 
so thank you for your help and for the link
Regards
